js/src/framework2.ts

```diff
@@ -44,6 +62,51 @@ export class Project {
     this.prompts = new PromptBuilder(this);
     this.scorers = new ScorerBuilder(this);
   }
+
+  public addPrompt(prompt: CodePrompt) {
+    this._publishablePrompts.push(prompt);
+    if (globalThis._lazy_load) {
+      globalThis._evals.prompts.push(prompt);
+    }
+  }
+
+  public addCodeFunction(
+    fn: CodeFunction<
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      any,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      any,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      GenericFunction<any, any>
+    >,
+  ) {
+    this._publishableCodeFunctions.push(fn);
+    if (globalThis._lazy_load) {
+      globalThis._evals.functions.push(fn);
+    }
+  }
+
+  async publish() {
```

Comment:
Do we need to clear up any of the `_publishableCodeFunctions` or `_publishablePrompts` on publish? 

I'm also wondering if we should avoid returning in this function. Technically we still have code functions pending, and it'll likely mean backward incompatible change, for example, if we decide to later return as well the created published code functions. 

I also wonder what this code will do if we have a `publish()` call within a file that also happens in a `push` (bundle) cli call.

Reply:
good call. i made it return if lazy loading is on (so it's a no-op)